forge-gui/res/cardsfolder/a/ali_from_cairo.txt

```diff
@@ -2,7 +2,7 @@ Name:Ali from Cairo
 ManaCost:2 R R
 Types:Creature Human
 PT:0/1
-R:Event$ LifeReduced | ActiveZones$ Battlefield | ValidPlayer$ You.lifeGE1 | Result$ LT1 | IsDamage$ True | ReplaceWith$ ReduceLoss | Description$ Damage that would reduce your life total to less than 1 reduces it to 1 instead.
+R:Event$ LifeReduced | ActiveZones$ Battlefield | ValidPlayer$ You.lifeGE1 | Amount$ LT1 | IsDamage$ True | ReplaceWith$ ReduceLoss | Description$ Damage that would reduce your life total to less than 1 reduces it to 1 instead.
```

Comment:
Lines 6 and 7 reference the parameter "Amount$" as of a change committed 2 days ago. Prior to the change two days ago, you would be correct.

6: SVar:ReduceLoss:DB$ ReplaceEffect | VarName$ Amount | VarValue$ X
7: SVar:X:ReplaceCount$Amount/LimitMax.Difference

Reply:
no, that's not how it works
Result$ is a param that gets checked by the Replacement Effect: https://github.com/Card-Forge/forge/blob/da0db2282c1c9625cb25c793bef9c222e64e3821/forge-game/src/main/java/forge/game/replacement/ReplaceLifeReduced.java#L48

L6+7 use a ReplacingObject which is an AbilityKey and not directly related

Reply:
Well, here's what I know. The version of the card that was changed as of the update two days ago (Check the changelog, please), broke the card such that it will trigger the replacement effect, but reduce the life anyway to zero.

I made that single change locally, and now the card works as intended.

It's quite possible I misunderstand the nuance of the code, but It did not work before, and with this change, it works again.

Reply:
then your problem is you're using new scripts with an old engine or vice versa
don't do that...

Reply:
I'm literally using the snapshot from a few hours ago. Fresh off the presses, dude. If you don't believe it's an issue, test the damn card on your side, but I'd prefer not to have to manually fix this card every time I download a snapshot.

It seems evident if you check the changelog:

https://github.com/Card-Forge/forge/commit/1bfa22e5b53b18bf6dcea512e3f59713065b75f9

That when Ali from Cairo was changed, they removed all references to a Result$ argument, and changed those to reference an Amount$ argument. They then left a Result$ argument in the effect code. Correcting this causes the card to work. 

Today's build. Not old code. Obviously.

Reply:
An alternative solution would be to change "VarName$ Amount" back to "VarName$ Result", which would probably to have the same effect. But the current state of the card does not function.

Reply:
by your logic all other cards there would require the same update
sorry, but it works fine 🤷‍♂️ 
![image](https://github.com/Card-Forge/forge/assets/8506892/b98b3df7-d5ca-444c-8065-54f7d04e2e9a)